core/Archive/ArchiveInvalidator.php

```diff
@@ -528,20 +535,48 @@ public function removeInvalidations($idSite, $plugin, $report = null)
     }
 
     /**
-     * Re-archives reports without propagating exceptions.
+     * Schedules a re-archiving reports without propagating exceptions. This is scheduled
+     * since adding invalidations can take a long time and delay UI response times.
      *
      * @param int|int[]|'all' $idSites
      * @param string $pluginName
      * @param string|null $report
      * @param Date|null $startDate
      */
-    public function reArchiveReportSafely($idSites, string $pluginName, string $report = null, Date $startDate = null)
+    public function scheduleReArchiving($idSites, string $pluginName, string $report = null, Date $startDate = null)
     {
         try {
-            $this->reArchiveReport($idSites, $pluginName, $report, $startDate);
+            $reArchiveList = new ReArchiveList($this->logger);
+            $reArchiveList->add([$idSites, $pluginName, $report, $startDate ? $startDate->getTimestamp() : null]);
```

Comment:
Personally would prefer if we used an array like `['idsites' => $idSites, ...]` this would allow us to quite easily extend or change the structure in the future if/when needed. and makes it maybe also more clear what each array item represents? Not too important though. It be mostly an issue if we eg ever removed for example an argument and then would need to do things like `[$idSites, $pluginName, null, null, $newArgument]` or something

Reply:
👍